src/contextify.cc

```diff
@@ -180,170 +182,201 @@ class ContextWrap : public ObjectWrap {
         return true;
     }
 
-    static NAN_PROPERTY_GETTER(GlobalPropertyGetter) {
-        NanScope();
-        Local<Object> data = args.Data()->ToObject();
+    static void GlobalPropertyGetter(Local<String> property, const PropertyCallbackInfo<Value>& info) {
+        Local<Object> data = info.Data()->ToObject();
         ContextifyContext* ctx = ObjectWrap::Unwrap<ContextWrap>(data)->ctx;
-        if (!ctx)
-            NanReturnUndefined();
-        Local<Value> rv = NanNew(ctx->sandbox)->GetRealNamedProperty(property);
-        if (rv.IsEmpty()) {
-            rv = NanNew(ctx->proxyGlobal)->GetRealNamedProperty(property);
+
+        if (!ctx) {
+            info.GetReturnValue().SetUndefined();
+            return;
         }
-        NanReturnValue(rv);
+
+        Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
+
+//        if (rv.IsEmpty()) {
+//            rv = Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property);
+//        }
+
+        info.GetReturnValue().Set(rv);
     }
 
-    static NAN_PROPERTY_SETTER(GlobalPropertySetter) {
-        NanScope();
-        Local<Object> data = args.Data()->ToObject();
+    static void GlobalPropertySetter(Local<String> property, Local<Value> value, const PropertyCallbackInfo<Value>& info) {
```

Comment:
It is not strictly necessary at the moment, if only targeting Node 0.12 up, 
however I believe that with some massaging of these things (and using the old 
Context::Enter for older Node versions with an ifdef, since I have not yet 
accounted for that in NAN), it could still be made to run on 0.10 and 0.8. I 
was mostly curious if there was some underlying issue with the wrappers which 
I had not foreseen.

On Tuesday 03 November 2015 01:55:57 Ido Ben-Yair wrote:
> >      }
> > 
> > -    static NAN_PROPERTY_SETTER(GlobalPropertySetter) {
> > -        NanScope();
> > -        Local<Object> data = args.Data()->ToObject();
> > +    static void GlobalPropertySetter(Local<String> property, Local<Value>
> > value, const PropertyCallbackInfo<Value>& info) {
> I agree that performance isn't the primary concern when using Contextify
> anyway. I'll try to convert it to the NAN version later on and tell you if
> it doesn't work again.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/brianmcd/contextify/pull/190/files#r43730428